### PR TITLE
Rewire the domain alias controller.

### PR DESCRIPTION
**Pull request by agentrickard**
Change the behavior of the DomainAliasListBuilder to be domain sensitive.

### CODE REVIEW

**Comment by artreaktor** [comment]
The object passed in DomainAliasController::listing
is of type DomainAliasListBuilder, not Domain
Getting Fatal error: Call to a member function id() on null in \DomainAliasListBuilder.php on line 77

**Comment by agentrickard** [comment]
Do you know how to fix that? 

See https://secure.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration

**Comment by agentrickard** [comment]
I also don't get that error. Did you rebuild the container? `drush cr`.

The route was rebuilt, and the variable should be passed from the routing call.

```
domain_alias.admin:
  path: '/admin/config/domain/alias/{domain}'
  defaults:
    _controller: '\Drupal\domain_alias\Controller\DomainAliasController::listing'
    _title: 'Domain aliases'
    _title_context: 'With modules'
  requirements:
    _permission: 'administer domains'
```

{domain} here should automatically be passed to listing().

File: domain_alias/src/Controller/DomainAliasController.php [hunk 1]
```diff
@@ -34,4 +35,21 @@ public function addAlias(DomainInterface $domain) {
...
+   * @param \Drupal\domain\DomainInterface $domain
+   *   An domain record entity.
+   *
+   * @return array
+   *   A render array as expected by drupal_render().
+   */
+  public function listing(DomainInterface $domain) {
+    $type = \Drupal::entityTypeManager()->getDefinition('domain_alias');
+    $storage = \Drupal::entityTypeManager()->getStorage('domain_alias');
+    $list = new DomainAliasListBuilder($type, $storage);
```
Comment (agentrickard) [inline domain_alias/src/Controller/DomainAliasController.php]: We may be able to just call the service directly. e.g.

```
$list = $this->entityManager()->getListBuilder($entity_type);
```

See EntityListController::listing().

**Pull request merged**